### PR TITLE
Performance investigation redis session

### PR DESCRIPTION
**Pull request by JamesGardiner**
Adds performance investigation results for the use of Redis to store EQ Questionnaire Runner session data.

See:
	https://github.com/ONSdigital/eq-questionnaire-runner/pull/90

for the reference implementation.

### CODE REVIEW

**Comment by JamesGardiner** [comment]
[Improve eq-session performance (L)](https://trello.com/c/22Y4UrcW/3593-improve-eq-session-performance-l)

File: doc/performance-investigations/0003-session-performance/summary.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+| 25, 25 | 462 | 416 | 1200 |
+| 50, 50 | 928 | 997 | 1600 |
+
+### Investigation
+
+| Clients & hatch rate | GET Average (ms) | POST Average (ms) | All requests average (ms) |
+| 1, 1 | 95 | 110 | 490 |
+| 5, 5 | 121 | 141 | 280 |
+| 25, 25 | 353 | 422 | 570 |
+| 50, 50 | 387 | 451 | 550 |
```
Comment (ajmaddaford) [inline doc/performance-investigations/0003-session-performance/summary.md:60]: The two tables are missing the table header dividing row (`|---|---|`), so don't render as tables.

File: run.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
 echo "Running Locust benchmark"
-USER_WAIT_TIME_MIN_SECONDS=0 USER_WAIT_TIME_MAX_SECONDS=0 REQUESTS_JSON="${1}" HOST="${2:-http://localhost:5000}" pipenv run locust --no-web -c 1 -r 1 -t ${3:-1m} --csv=${4:-output} -L WARNING
+USER_WAIT_TIME_MIN_SECONDS=0 USER_WAIT_TIME_MAX_SECONDS=0 REQUESTS_JSON="${1}" HOST="${2:-http://localhost:5000}" pipenv run locust --no-web -c 50 -r 50 -t ${3:-1m} --csv=${4:-output} -L WARNING
```
Comment (ajmaddaford) [inline run.sh:4]: This shouldn't be committed.

File: doc/performance-investigations/0003-session-performance/baseline1Client_stats_history.csv [hunk 3]
```diff
@@ -0,0 +1,32 @@
+"Type","Name","Timestamp","# requests","# failures","Requests/s","Requests Failed/s","Median response time","Average response time","Min response time","Max response time","Average Content Size","50%","66%","75%","80%","90%","95%","98%","99%","99.9%","99.99%","99.999","100%"
```
Comment (ajmaddaford) [inline doc/performance-investigations/0003-session-performance/baseline1Client_stats_history.csv:1]: The file naming is inconsistent, they should be lowercase and underscore separated. This file should be `baseline_1_client_stats_history.csv`.

File: doc/performance-investigations/0003-session-performance/summary.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
+# Use Redis for storing Questionnaire Runner sessions
+
+The EQ session data are currently stored in Datastore. It may be more performant to store these data in Redis (which is already used for JWT).
```
Comment (ajmaddaford) [inline doc/performance-investigations/0003-session-performance/summary.md:3]: - This should be either "EQ sessions are currently" or "EQ session data is currently".
- "store these data" -> "store this data"

**Comment by JamesGardiner** [comment]
[Use Redis to store EQ Session data](https://trello.com/c/5qfIBhMf/3798-use-redis-to-store-eq-session-data)

**Comment by JamesGardiner** [comment]
[Use Redis to store EQ Session data](https://trello.com/c/5qfIBhMf/3798-use-redis-to-store-eq-session-data)

**Comment by JamesGardiner** [comment]
[Use Redis to store EQ Session data](https://trello.com/c/5qfIBhMf/3798-use-redis-to-store-eq-session-data)

**Pull request merged**